src/FRP/Event/Class.purs

```diff
@@ -0,0 +1,62 @@
+module FRP.Event.Class
+  ( class IsEvent
+  , fold
+  , folded
+  , count
+  , mapAccum
+  , mapMaybe
+  , withLast
+  , sampleOn
+  , sampleOn_
+  ) where
+
+import Prelude
+
+import Control.Alternative (class Alternative)
+import Data.Maybe (Maybe(..))
+import Data.Monoid (class Monoid, mempty)
+import Data.Tuple (Tuple(..), snd)
+
+-- | Functions which an `Event` type should implement, so that
+-- | `Behavior`s can be defined in terms of any such event type:
+-- |
+-- | - `fold`: combines incoming values using the specified function,
+-- | starting with the specific initial value.
+-- | - `mapMaybe`: discards incoming values which do not satisfy a predicate.
+-- | - `sampleOn`: samples an event at the times when a second event fires.
+class Alternative event <= IsEvent event where
```

Comment:
I really like this; wondering what else I can do with it...

Reply:
Well, you could probably create instances for other types of `Event`, like the ones from existing JS libraries 😄